activerecord/lib/active_record/destroy_association_later_job.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  class DestroyAssociationLaterError < StandardError
+  end
+
+  # Destroy record association in a background job.
+  #
+  #  class ParentNode < ApplicationRecord
+  #    has_one :child, dependent: :destroy_later
+  #   end
+  #
+  #   +destroy_later+ param adds an +after_destroy+ callback that schedules an ActiveRecord::DestroyAssocitionLaterJob.
+  #
+  class DestroyAssociationLaterJob < ActiveJob::Base
+    queue_as { ActiveRecord::Base.queues[:destroy] }
+
+    discard_on ActiveJob::DeserializationError
+
+    def perform(owner_model_name: nil, owner_id: nil, assoc_class: nil,
```

Comment:
Why are we passing the class names and ids and not the Active Record instance itself and use globalid to fetch the right records?

Reply:
Ah. Just saw in the original thread the reason for that.